File: LeetCode/Easy/1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.java
class Solution {
    public int[] finalPrices(int[] prices) {
        int n = prices.length;
        int[] ret = new int[n];

        for (int i = 0; i < n; ++i){
            int v = prices[i];
            for (int j = i + 1; j < n; ++j)
                if (prices[j] <= prices[i]){
                    v -= prices[j];
                    break;
                }
            ret[i] = v;
        }
        
        return ret;
    }
}

File: LeetCode/Easy/1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.go
func finalPrices(prices []int) []int {
    n := len(prices)
    ret := make([]int, n)

    for i := 0; i < n; i++ {
        v := prices[i]
        for j := i + 1; j < n; j++ {
            if prices[j] <= prices[i] {
                v -= prices[j]
                break;
            }
        }
        ret[i] = v
    }

    return ret
}